flocker/control/httpapi.py

```diff
@@ -220,8 +222,7 @@ def create_dataset_configuration(self, primary, dataset_id=None,
         )
         manifestation = Manifestation(dataset=dataset, primary=True)
 
-        primary_node = deployment.get_node(ip_to_uuid(primary),
-                                           hostname=primary)
+        primary_node = deployment.get_node(primary)
```

Comment:
I find it really confusing that `deployment.get_node` should return a new `Node` if the supplied uuid isn't among the current nodes....but it looks like it was always like that and the hostname would only ever have been used if a new Node was created.